.circleci/workflow-queue-v1.sh

```diff
@@ -0,0 +1,110 @@
+#! /bin/bash
+
+set -e
+set -o pipefail
+set -o functrace
+
+#********************
+# VARIABLES
+# *******************
+api_root="https://circleci.com/api/v1/"
+# polling branch name. Other branches (PRs) are excluded from consideration.
+branch="master"
+# Polling workflow name. Other workflow names (nightly-tests and deploy-staging) are excluded from consideration.
+workflow_name="build-test-deploy"
+project_slug="all-of-us/workbench"
+
+# Max check time on a workflow is 45 minutes because e2e tests may take a long time to finish.
+# DISCLAIMER This max time may not be enough.
+max_time=$((45 * 60))
+sleep_time=30
+
+#********************
+# FUNCTIONS
+# *******************
+
+# CIRCLECI_API_TOKEN is a env variable whose value is project token.
+# See https://circleci.com/docs/2.0/managing-api-tokens/#creating-a-personal-api-token
+check_circleci_token() {
+  if [[ ! $CIRCLECI_API_TOKEN ]]; then
+    printf '%s\n' "Required env variable \"CIRCLECI_API_TOKEN\" is not found." >&2
+    exit 1
+  fi
+}
+
+check_circleci_build_num() {
+  if [[ ! $CIRCLE_BUILD_NUM ]]; then
+    printf '%s\n' "Required env variable \"CIRCLE_BUILD_NUM\" is not found." >&2
+    exit 1
+  fi
+}
+
+check_circleci_workflow_id() {
+  if [[ ! $CIRCLE_WORKFLOW_ID ]]; then
+    printf '%s\n' "Required env variable \"CIRCLE_WORKFLOW_ID\" is not found." >&2
+    exit 1
+  fi
+}
+
+circle_get() {
+  local url="${api_root}${1}"
+  printf "curl GET ${url}\n" >/dev/tty
+  curl -X GET -s -S -f \
+    -H "Content-Type: application/json" \
+    -H "Circle-Token: ${CIRCLECI_API_TOKEN}" \
+    "${url}"
+}
+
+# Fetch list of builds on master branch that are running, pending or queued.
+fetch_pipelines() {
+  printf '%s\n' "Fetch previously submitted builds on \"${branch}\" branch that are running, pending or queued:"
+  local get_path="project/${project_slug}?filter=running&shallow=true"
+  local get_result=$(circle_get "${get_path}")
+  if [[ ! "${get_result}" ]]; then
+    printf "curl failed."
+    exit 1
+  fi
+  jq_filter="select(.branch==\"${branch}\" and .build_num < $CIRCLE_BUILD_NUM and (.status | test(\"running|pending|queued\"))) | select(.workflows.workflow_name==\"${workflow_name}\" and .workflows.workflow_id!=\"${CIRCLE_WORKFLOW_ID}\")"
```

Comment:
nit: should probably line wrap for readability

Reply:
👍🏻